src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -142,7 +147,8 @@ public ExpressionGraph() {
    */
   public void addSubExpressionGraph(ExpressionGraph subGraph,
                                     ExpressionEdge mainGraphInput, ExpressionVertex subGraphInput,
-                                    ExpressionEdge mainGraphOutput, ExpressionVertex subGraphOutput) {
+                                    ExpressionEdge mainGraphOutput, ExpressionVertex subGraphOutput,
+                                    Map<VariableReferenceVertex, VariableReferenceVertex> variableRenamingMap) {
```

Comment:
I think we want Map<VariableIdentifier, VariableIdentifier> here right? Adding a variable reference vertex to the main graph also puts it in the VariableIdentifier -> VariableReferenceVertex map in ExpressionGraph, which in this case it isn't doing.

Unless those vertices have already been added to the main graph via. the addVertex function earlier.